src/compartment.c

```diff
@@ -53,15 +53,20 @@ entry_point_cmp(const void* val1, const void* val2)
  * a `struct Compartment`. At this point, we only read data.
  */
 struct Compartment*
-comp_from_elf(char* filename, char** entry_points)
+comp_from_elf(char* filename, struct ConfigEntryPoint* entry_points, size_t entry_point_count)
 {
     struct Compartment* new_comp = comp_init();
     int pread_res;
+
+    // TODO move somewhere else?
+    // TODO we can potentially handle object files, so don't need to explicitly
+    // check for `main`
     if (!entry_points)
     {
-        char* main_entry[1] =  { "main" };
-        entry_points = main_entry;
+        entry_points = NULL;
```

Comment:
> I'm not sure how this relates to purecap vs hybrid

Oh, because in purecap it's _not_ a no-op: `!entry_points` only considers the address field, so you could pass in an authentic capability with an address of zero, and we'd squash it to the canonical `NULL`. I doubt that it matters for normal uses, but I thought that the canonicalisation might be useful for some reason.

Reply:
Good point; I hadn't considered that, so that's not the case here. I just want to ensure `entry_points` is not `NULL`. But now it does raise the question if in purecap, for a pointer `p`, is `!p` still equivalent to `p != NULL`?

Reply:
I think `!p` is the same as `p != NULL`, yes. At least empirically, they compile to the same code, which just checks the address.

As a quick check: [C99](https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) says that the result of `!` depends on whether the argument compares equal to zero (i.e. `(0==p)` in this case). Similar wording applies to simple conversion to `_Bool` as well as interpretation in an `if` statement. The only notable difference from comparison with `NULL` is that `NULL` might not be defined to be simply `0`.

Things are slightly different in C++, which (amongst other differences) is [more strict about what `NULL` can be](https://en.cppreference.com/w/cpp/types/NULL), but I think the behaviour is the same.